### PR TITLE
Rewrite prototype timetree layout using d3 force

### PR DESCRIPTION
**Pull request by rlskoeser**

### CODE REVIEW

**Comment by rlskoeser** [comment]
@quadrismegistus not sure what kind of code review is useful here, but would appreciate you taking a quick look. Ignore the old code in the diff. Any comments on code readability or logic or next steps probably useful.

File: assets/js/timetree.js [hunk 1]
```diff
@@ -1,187 +1,261 @@
...
+        // cast all to numeric to avoid duplication
+        branches[leaf.branch].add(Number(leaf.century));
+      });
+
+      // create a node for the trunk
+      nodes.push({
+        id: 'trunk',
+        title: 'trunk',
+        type: 'trunk'
+      });
```
Comment (quadrismegistus) [inline assets/js/timetree.js:53]: 🤣 this is awesome

File: assets/js/timetree.js [hunk 2]
```diff
@@ -1,187 +1,261 @@
...
+        .attr("id", d => "c" + d)
+        .attr("height", leafContainerHeight)
+        .attr("width", width)
+        .attr("x", min_x)
+        .attr("y", (d, i) => (min_y + i * leafContainerHeight))
+        .attr('fill', "lightgray")
+        .attr('fill-opacity', 0.1)
+        .attr('stroke', 'gray')
+        .attr('stroke-opacity', 0.5)
+        .attr('stroke-dasharray', 2);
```
Comment (quadrismegistus) [inline assets/js/timetree.js:144]: So these are the ghost nodes?

File: assets/js/timetree.js [hunk 3]
```diff
@@ -1,187 +1,261 @@
...
+      top:  localTop,
+      bottom: localTop + leafContainerHeight
+    };
+  });
+
+  // draw a couple of lines to help gesture at tree-ness
+  let trunkWidth = 65;
+  // right side
+  background.append("path")
+    .attr("d", d3.line().curve(d3.curveNatural)([
```
Comment (quadrismegistus) [inline assets/js/timetree.js:171]: Could we be using these curved paths throughout the tree?

**Review by quadrismegistus — APPROVED** [review]
Looks great! All my comments are really thoughts for future iterations. This PR looks great!

File: assets/js/timetree.js (hunk 2 above)
Comment (rlskoeser) [reply assets/js/timetree.js:144]: These "leaf containers" are rectangular bands to position things roughly by century — I don't love the styling and probably will shift some of it to CSS, but it seemed good enough for an initial version. Also, in the current layout the leaves aren't strictly contained within those rectangles, so we'll need to either tweak the layout or tweak the century display. Hoping to hold off on that until Gissoo can review and advise.

File: assets/js/timetree.js (hunk 3 above)
Comment (rlskoeser) [reply assets/js/timetree.js:171]: yes! that is my thought for one of the next steps  — I want to draw branches over the network graph to connect the trunk to the leaves. Thinking we could implement a method that introduces some amount of variation between the starting point and ending point to make the lines look a little more natural and tree-like.

**Pull request merged**